coalib/parsing/LineParser.py

```diff
@@ -82,6 +82,15 @@ def _parse(self, line):
             line,
             self.comment_separators)
         comment = unescape(comment)
+
+        if '#' in comment:
+            comment_index = comment.index('#')
+            if comment_index + 1 < len(comment) or comment_index - 1 > 0:
```

Comment:
You check if `comment_index + 1` is within bounds OR `comment_index - 1` is within bounds. So even if one of the two passes, we'll enter the conditional branch. But the next line uses both `comment[comment.index('#') + 1]` and `comment[comment.index('#') - 1]`. This will definitely raise an exception at the extremes. Let's just have two separate conditions.